### PR TITLE
[breaking]upgrade to dotnet core 3.1

### PR DESCRIPTION
**Pull request by live1206**
Pair PR: https://ceapex.visualstudio.com/Engineering/_git/Docs.Build/pullrequest/19461

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5414)

### CODE REVIEW

File: src/docfx/docfx.csproj [hunk 1]
```diff
@@ -18,21 +18,20 @@
...
     <PackageReference Include="Microsoft.ChakraCore" Version="1.11.15" />
-    <PackageReference Include="Microsoft.DocAsCode.MarkdigEngine.Extensions" Version="2.48.0-b-191110006-ga90f97" />
-    <PackageReference Include="Microsoft.Graph" Version="1.20.0" />
-    <PackageReference Include="Microsoft.Identity.Client" Version="4.7.0" />
+    <PackageReference Include="Microsoft.DocAsCode.MarkdigEngine.Extensions" Version="2.48.0" />
+    <PackageReference Include="Microsoft.Graph" Version="1.21.0" />
+    <PackageReference Include="Microsoft.Identity.Client" Version="4.7.1" />
     <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
     <PackageReference Include="Jint" Version="2.11.58" />
+    <!-- To be compatible with Microsoft.Docs.DocumentHosting.RestClient in docs-pipeline -->
```
Comment (live1206) [inline src/docfx/docfx.csproj]: https://dev.azure.com/ceapex/Engineering/_workitems/edit/158883?src=WorkItemMention&src-action=artifact_link

**Comment by live1206** [comment]
image
Fails when the entry project is `docfx.exe`, but the same code works when the entry project is `docfx.Test`

File: src/docfx/template/RazorTemplate.cs [hunk 2]
```diff
@@ -16,6 +16,9 @@
 using Microsoft.AspNetCore.Mvc.Controllers;
 using Microsoft.AspNetCore.TestHost;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+[assembly: ApplicationPart("Microsoft.Docs.Template.Views")]
```
Comment (yufeih) [inline src/docfx/template/RazorTemplate.cs]:
```suggestion
[assembly: ApplicationPart("Microsoft.Docs.Template")]
```

Comment (yufeih) [reply src/docfx/template/RazorTemplate.cs]: This will do the trick!

**Comment by live1206** [comment]
- The HTML diff(space and new line) of `index.raw.page.json` is acceptable since HTML will be normalized during migration CI run.
- The blob diff should not matter either since there is no content diff related
- The diff for `DevSandbox` looks like CRR content update

**Comment by yufeih** [comment]
`DevSandbox` diff might be related to markdig engine upgrade, `:::code` is a new syntax that was recently added

**Comment by live1206** [comment]
> `DevSandbox` diff might be related to markdig engine upgrade, `:::code` is a new syntax that was recently added

Yes, according to https://github.com/dotnet/docfx/pull/5416#issuecomment-570462364. 
But

**Pull request merged**